### ISSUE TITLE
Menu: Focusing menu items on mouseenter

### ISSUE THREAD

**Issue by ling1726** [microsoft/fluentui#18319]
For screen reader users, the mouse cursor can be anywhere on the page. If the `Menu` by chance opens into the mouse cursor it is confusing for screen reader users since it can:
* steal focus
* open a nested menu

The requirement that mouse hover and focus should have the same state in the menu. Figure out more a11y friendly ways to achieve this

**Comment by ling1726** [comment]
So after a bit of consideration I can come to the conclusion that determining a coherent behaviour for both keyboard+mouse users is incredibly difficult since there are so many factors to verify if a user is completely or partially using a keyboard.

Looking at the progress the WAI are having with their Menu implementation:
* w3c/aria-practices#598
* w3c/aria-practices#593

The same conclusion has been reached there that it is more beneficial to set mouse focus on hover. I won't reiterate the arguments, feel free to take a look at that issue+PR the discussion is quite interesting.

In Fluent we can settle on a compromise since our nested popover submenus are really something that is not seen often in web.  I propose that we:

* Set focus on hover
* Do not open menus/submenus on hover unless a mousemove event is detected

Relying on mouse move gives us a more accurate picture that a user is using the mouse as opposed to tracking how items are focused. In this case appropriate throttling would limit performance impact using the mouse move event.

FYI @smhigley  @jurokapsiar
